### PR TITLE
fix: clippy unused

### PR DESCRIPTION
**Pull request by gruberb**
# Description

Remove `unused = "allow"` from the `Cargo.toml` to highlight unused imports and variables. Cleaning up the warnings.

## PR Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/topos/tests/utils.rs [hunk 1]
```diff
@@ -37,6 +40,8 @@ pub async fn setup_polygon_edge(path: &str) -> String {
     installation_path.to_str().unwrap().to_string()
 }
 
+// Have to allow dead_code because clippy doesn't recognize it is being used in the tests
+#[allow(dead_code)]
```
Comment (Freyskeyd) [inline crates/topos/tests/utils.rs:44]: does using `#[cfg(test)]` could work?

Comment (dvdplm) [reply crates/topos/tests/utils.rs:44]: Hmm. How can the compiler think it's dead code if it's `pub`? I assume the module is `pub` too?

**Review by dvdplm — APPROVED** [review]
LGTM modulo the question about dead code.

File: crates/topos/tests/utils.rs (hunk 1 above)
Comment (gruberb) [reply crates/topos/tests/utils.rs:44]: No, it didn't... I can try with other flags or dig deeper into why!

File: crates/topos-tce-api/tests/grpc/certificate_precedence.rs [hunk 2]
```diff
@@ -22,12 +20,12 @@ async fn fetch_latest_pending_certificates() {
     let validator_store: Arc<ValidatorStore> =
         create_validator_store(vec![], futures::future::ready(fullnode_store.clone())).await;
 
-    let (tx, rx): (
+    let (_, _): (
```
Comment (Freyskeyd) [inline crates/topos-tce-api/tests/grpc/certificate_precedence.rs:23]: I think that it can be completely removed, right?

**Review by Freyskeyd — APPROVED** [review]

**Review by atanmarko — APPROVED** [review]

File: crates/topos-tce-storage/src/types.rs [hunk 3]
```diff
@@ -56,7 +56,7 @@ pub struct CheckpointSummary {
     sequence_number: usize,
     checkpoint_data: Vec<SourceStreamPosition>,
 }
-pub struct VerifiedCheckpointSummary(CheckpointSummary, ValidatorQuorumSignatureInfo);
+pub struct VerifiedCheckpointSummary((), ());
```
Comment (gruberb) [inline crates/topos-tce-storage/src/types.rs:59]: Seems like these are not being used at all @Freyskeyd. Is there a reason for it?

**Pull request merged**